contracts/Spades.sol

```diff
@@ -113,65 +151,135 @@ contract Spades {
     // txNonce incresases by one;
 
         txMap[txNonce] = transaction;
-        whoSigned[txNonce][msg.sender] = true;
+        whoSignedTx[txNonce][msg.sender] = true;
         txNonce ++;
 
         emit Submit(msg.sender, _amount, txNonce, _data);
 
     }
     
-        // Returns a transaction when given the tx_index
-    function getTransaction(uint txIndex) public view returns (address to, uint amount, uint confirmations, address signature) {
+        // Returns a transaction when given the tx_index.
+    function getTransaction(uint txIndex) public view returns (address targetAccount, uint amount, uint confirmations, address signature, bool executed) {
     
-        Transaction memory transaction = txMap[txIndex];
+        Transaction storage transaction = txMap[txIndex];
```

Comment:
```suggestion
        transaction = txMap[txIndex];
```
Won't this compile like this?

Reply:
This won't compile. You need to specify `storage`because memory is the default.

Reply:
Your suggestion doesn't work, it will assume `memory` as default. We need to specify `storage`.